### DISCUSSION TITLE
Touch values

### DISCUSSION THREAD

**Discussion by mohammad-torabi-84** [moononournation/Arduino_GFX#336] [Q&A]
Hello,

in the touch.h file, what is the value for the following variables? for the ESP32-8048S070C board I mean.

Thanks,

// Please fill below values from Arduino_GFX Example - TouchCalibration
bool touch_swap_xy = false;
int16_t touch_map_x1 = -1;
int16_t touch_map_x2 = -1;
int16_t touch_map_y1 = -1;
int16_t touch_map_y2 = -1;

**Comment by moononournation** [comment]
If you are using Capacitive Touch screen, you can leave it -1. Or you can use TouchCalibration.ino example to check the correct value.

**Reply by mohammad-torabi-84** [reply]
the touch_touched() function always returns true, no matter the display touched or not.

**Reply by moononournation** [reply]
you should config the touch by uncomment you touch chip #define

**Comment by mohammad-torabi-84** [comment]
based on the schematics it is using XPT2046, but it keeps calling the touch_touched() function.

**Reply by moononournation** [reply]
yes, it keeps calling for every loop.

**Reply by mohammad-torabi-84** [reply]
this is the Serial output when the `Serial.printf("touch_raw_x: %d, touch_raw_y: %d, count: %d\n", touch_raw_x, touch_raw_y, count);` is uncommented. please note that the display was touched.

20:21:56.473 -> touch_raw_x: -4096, touch_raw_y: 8191, count: 5309
20:21:56.473 -> touch_raw_x: -4096, touch_raw_y: 8191, count: 5310
20:21:56.473 -> touch_raw_x: -4096, touch_raw_y: 8191, count: 5311
20:21:56.473 -> touch_raw_x: -4096, touch_raw_y: 8191, count: 5312
20:21:56.473 -> touch_raw_x: -4096, touch_raw_y: 8191, count: 5313
20:21:56.473 -> touch_raw_x: -4096, touch_raw_y: 8191, count: 5314
20:21:56.473 -> touch_raw_x: -4096, touch_raw_y: 8191, count: 5315
20:21:56.506 -> touch_raw_x: -4096, touch_raw_y: 8191, count: 5316
20:21:56.506 -> touch_raw_x: -4096, touch_raw_y: 8191, count: 5317
20:21:56.506 -> touch_raw_x: -4096, touch_raw_y: 8191, count: 5318
20:21:56.506 -> touch_raw_x: -4096, touch_raw_y: 8191, count: 5319
20:21:56.506 -> touch_raw_x: -4096, touch_raw_y: 8191, count: 5320
20:21:56.506 -> touch_raw_x: -4096, touch_raw_y: 8191, count: 5321
20:21:56.506 -> touch_raw_x: -4096, touch_raw_y: 8191, count: 5322
20:21:56.506 -> touch_raw_x: -4096, touch_raw_y: 8191, count: 5323

as you can see

**Reply by moononournation** [reply]
What configuration?

**Reply by mohammad-torabi-84** [reply]
Board: ESP32-8048S070C
touch driver: based on schematic- XPT2046
I uncomment the define section on touch.h file and run the TouchCalibration on  (https://github.com/moononournation/Arduino_GFX/tree/master/examples/TouchCalibration). just uncomment the Serial.println line in the TouchCalibration.ino file to see the output.

**Reply by moononournation** [reply]
I mean can you show what you uncommented and what changed in your test?

**Comment by mohammad-torabi-84** [comment]
in the touch.h file, I uncomment the define section for XPT2046 as below
/*uncomment for XPT2046 */
#define TOUCH_XPT2046
#define TOUCH_XPT2046_SCK 12
#define TOUCH_XPT2046_MISO 13
#define TOUCH_XPT2046_MOSI 11
#define TOUCH_XPT2046_CS 10
#define TOUCH_XPT2046_INT 18
#define TOUCH_XPT2046_ROTATION 0
#define TOUCH_XPT2046_SAMPLES 50

and in TouchCalibration.ino in the beginning of the loop function as below.

  if (touch_touched()) {
    int32_t total_x = touch_raw_x, total_y = touch_raw_y;
    int count = 1;
    while (touch_touched()) {
      total_x += touch_raw_x;
      total_y += touch_raw_y;
      count++;
      Serial.printf("touch_raw_x: %d, touch_raw_y: %d, count: %d\n", touch_raw_x, touch_raw_y, count);
    }
    touched_x[current_point] = total_x / count;
    touched_y[current_point] = total_y / count;
    Serial.printf("touched_x: %d, touched_y: %d\n", touched_x[current_point], touched_y[current_point]);
.
.
.

**Comment by moononournation** [comment]
If the connection ok, the calibration output should be something like this:
```
Touch Calibration
Init display
touched_x[0]: 659, touched_y[0]: 559
touched_x[1]: 3262, touched_y[1]: 693
touched_x[2]: 611, touched_y[2]: 3483
touched_x[3]: 3414, touched_y[3]: 3432
bool touch_swap_xy = false;
int16_t touch_map_x1 = 185;
int16_t touch_map_x2 = 3788;
int16_t touch_map_y1 = 155;
int16_t touch_map_y2 = 3928;
```
